alf/experience_replayers/replay_buffer.py

```diff
@@ -32,21 +100,216 @@ def __init__(self,
                  max_length=1024,
                  device="cpu",
                  allow_multiprocess=False,
+                 her_k=0,
```

Comment:
It seems to me HER should be subclass or a wrapper of ReplayBuffer. ReplayBuffer is supposed to be more general.

Reply:
I can subclass as HERReplayBuffer and have all experience_replayers use that instead of ReplayBuffer, since HERReplayBuffer is efficient when k=0.

Will that be Ok?

Reply:
We are going to have PrioritizedReplayBuffer. We need to a way to flexibly combine these things:
plain replay buffer
HER with plain replay buffer
prioritized replay buffer
HER with prioritized replay buffer

Reply:
Then having two flags controlling HER and Prioritized replay is probably the best solution at the moment.  Prioritization can be done in this function: _get_sample_batch, which is independent of HER.  If things become even more complex, then maybe we could use a chain of inheritance ReplayBuffer=>HERReplayBuffer=>PrioritizedHERReplayBuffer, and have experience_replayers use the most capable replay_buffer.